web/web/models/meter.py

```diff
@@ -81,41 +113,51 @@ def get_interval_count(self, start, end):
         if end == None:
             end = datetime.now()
 
-        for interval in self_intervals:
-            if interval.start_time >= start and interval.end_time <= end:
-                selected_intervals.append(interval)
+        for meter_interval in self_intervals:
+            if meter_interval.start_time >= start and meter_interval.end_time <= end:
+                selected_intervals.append(meter_interval)
 
         return len(selected_intervals)
 
     def get_rates(self):
         '''Returns meter instance's rates as a list'''
 
         rates = []
-        for interval in self.intervals:
-            if interval.rate.description not in rates:
-                rates.append(interval.rate.description)
-
+        for meter_interval in self.meter_intervals:
+            if meter_interval.rate.description not in rates:
+                rates.append(meter_interval.rate.description)
+        
         return rates
 
     def get_channels(self):
         '''Returns meter instance's channel settings as a list'''
-
-        channels = []
-        for channel in self.channels:
-            if channel.setting not in channels:
-                channels.append(channel.setting)
-
+        
+        channels = Meter.query.\
+                        join(Meter.channels). \
+                        all()
         return channels
 
     def get_all_intervals(self):
         '''Returns all meter instances's intervals in a list'''
 
         intervals_list = []
-        for interval in self.intervals:
-            intervals_list.append(interval.interval_id)
-
+        for meter_interval in self.meter_intervals:
+            intervals_list.append(meter_interval.meter_interval_id)
+        
         return intervals_list
 
+    @classmethod
+    def get_pv_meters(cls):
+        '''Returns all meter that are pvs'''
```

Comment:
Can you explain this method a bit more. It seems like we are filtering all `Meters` to find all that are PV, and PV is determined by Channel Type?

Reply:
I guess why not put this prop on the meter model?

Reply:
I just realized that the PV model has a FK to meter. This function should be able to find all meters that are PV from that association.

Reply:
This meter class method is intended to be used for the initial set up of creating the relationships for which pvs are meters. I used it to set up my local database in order to test out the api routes, I thought it would make it easier for which this is set up later on.  Yes, meters that are PV are ones that have the channel type ('R') which is reverse flow direction. Let me know if you think this is useful!

Reply:
That relationship is already established on the PV model via the foreign key right? I think we need to re-work this function or get rid of it if all that is needed is to get all PVs that have a non-null FK to meter. 

Or maybe I'm missing something....can you provide an example of how this is used?

Also, this function seems to guarantee to return all meters that are PV, however it technically only returns all meters that have a Channel Type of `R`.  Which can make this brittle.